Inform user about resampled fitting range

moritz-gerster opened the pull request:
### What is the problem?
IRASA works by resampling the time series for different h-values of a given h-set. The oscillatory peaks are shifted up and down and can thus be eliminated. This resampling procedure can be quite dangerous though if frequency ranges of the PSD are evaluated which should not be evaluated. This corresponds to challenge 1 (page 16) in [1]. 

For example, when using a 1 Hz highpass filter, one should not use a fitting range from 1-100Hz because using the default h-set (with the maximum resampling factor being ca. h_max=2), a frequency range of ca. 0.5-200Hz will be evaluated. This includes the highpass filtered range which should be avoided. See Fig. 5 b) in [1].

The equation to calculate the evaluated fitting range (in contrast to the fitting range) is given on page 17 of [1]. 

### What is the solution?
I inserted a few lines of code that explicitly inform the user about the evaluated fitting range. That might be enough to give the user a better idea of which part of the PSD IRASA actually evaluates. 

### Reference
[1] Separating neural oscillations from aperiodic 1/f activity: challenges and recommendations
Moritz Gerster, Gunnar Waterstraat, Vladimir Litvak, Klaus Lehnertz, Alfons Schnitzler, Esther Florin, Gabriel Curio, Vadim Nikulin
bioRxiv 2021.10.15.464483; doi: https://doi.org/10.1101/2021.10.15.464483

raphaelvallat commented:
Hi @moritz-gerster,

Thank you for opening the issue and adding your expert knowledge to this function. I will make sure to read your preprint. The PR looks great. A few naive questions:

1. Should we raise an explicit warning when the `band` parameter is ill-defined, as in the example you describe? Although this would require knowing the filtering operations that were previously applied to the data, which we might be able to get from the MNE Raw structure. 
2. Could the initial sampling frequency of the data lead to a similar issue (e.g. if it's too low)?
3. In sleep research, it is common to bandpass filter the data between 0.5-35 Hz and downsample the data to 100 Hz. Based on that, which optimal `band` parameter would you recommend? Is the current 1-30 Hz appropriate?

As for the PR, my only comment is that instead of using `print`, we should use [logging](https://docs.python.org/3/howto/logging.html). This allows user to easily disable the print output with the `verbose` parameter. You will need to:

1) add a `verbose` parameter to the function, e.g: https://github.com/raphaelvallat/yasa/blob/f5d6afaa9cc877ad1f5073ee540eca7bc364d9ef/yasa/detection.py#L423-L427
2) use `logger.warning` instead of `print`

Cheers,
Raphael

moritz-gerster commented:
Dear @raphaelvallat,

These are very good points. I actually did think about raising warnings but initially, I did not want to mess around too much with your code.

1. I changed the print statement to logging.info. I further included logging.warning for the cases of too low or too high band parameters with regards to the applied filter settings. 
2. Yes, this is true too. For a sampling rate of 1000 Hz the Nyquist frequency is 500 Hz. However, with the default h-set with h_max=1.9, this would further reduce to 263 Hz. Therefore I added another warning.
3. A default fitting range of 1-30 Hz corresponds to an evaluated frequency range of 1/1.9=0.53 Hz to 30*1.9=57 Hz at the default h_max=1.9. Therefore, the data should not be downsampled below 114 Hz and should not be lowpass filtered below 114 Hz for this band. If you keep the filter settings at 0.5-35Hz, the default band should be as low as 1-18 Hz.

The logging.info and warnings are quite long and detailed now. One could consider combining them all into one warning printing the fitting range (band), the evaluated frequency range (band_evaluated), and, if known, the filter settings.

Thank you for introducing me to logging.

Best wishes,
Moritz
